File: entsoe-client/src/main/java/org/example/model/common/MarketRoleType.java
package org.example.model.common;

import java.util.Arrays;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum MarketRoleType {
  ISSUING_OFFICE("A32", "Issuing Office"),
  MARKET_OPERATOR("A33", "Market Operator");

  private final String code;
  private final String description;

  public static MarketRoleType fromCode(String code) {
    return Arrays.stream(values())
        .filter(type -> type.code.equals(code))
        .findFirst()
        .orElseThrow(() -> new IllegalArgumentException("Unknown MarketRoleType code: " + code));
  }
}

File: entsoe_client/src/entsoe_client/model/common/market_role_type.py
from enum import Enum
from typing import Self

from entsoe_client.exceptions.unknown_market_role_type_error import (
    UnknownMarketRoleTypeError,
)


class MarketRoleType(Enum):
    ISSUING_OFFICE = ("A32", "Issuing Office")
    MARKET_OPERATOR = ("A33", "Market Operator")

    def __init__(self, code: str, description: str) -> None:
        self.code = code
        self.description = description

    @classmethod
    def from_code(cls, code: str) -> Self:
        for member in cls:
            if member.code == code:
                return member
        raise UnknownMarketRoleTypeError(code)
